hswidget/fonttablewidget/state.go

```diff
@@ -0,0 +1,80 @@
+package fonttablewidget
+
+import (
+	"fmt"
+
+	"github.com/ianling/giu"
+)
+
+type fontTableWidgetMode int
+
+const (
+	// viewer state
+	fontTableWidgetViewer fontTableWidgetMode = iota
+	// edit rune
+	fontTableWidgetEditRune
+	// opens a dialog and adds a new item on first free index
+	fontTableWidgetAddItem
+)
+
+type widgetState struct {
+	mode     fontTableWidgetMode
+	editRune editRune
+	addItem  addItem
+}
+
+// Dispose cleans state
+func (s *widgetState) Dispose() {
+	s.editRune.Dispose()
+	s.addItem.Dispose()
+}
+
+type editRune struct {
+	editedRune int32
+	startRune  rune
+}
```

Comment:
`editRune` sounds like an action instead of a thing. would this make more sense as `runeEditState` ?

Reply:
solved by 7d29fa5